internal/reminder/reminder.go

```diff
@@ -144,19 +147,64 @@ func (r *reminder) sendReminders(ctx context.Context) []error {
 
 	logger.Info().Msgf("created repository batch of size: %d", len(repos))
 
-	// Update the reminder_last_sent for each repository to export as metrics
+	messages := make([]*message.Message, 0, len(repos))
+	errorSlice := make([]error, 0)
+
+	tx, err := r.store.BeginTransaction()
+	if err != nil {
+		return []error{err}
+	}
+
+	defer tx.Rollback()
+
+	qtx := r.store.GetQuerierWithTransaction(tx)
+
+	// TODO: Collect Metrics
+	// Potential metrics:
+	// - Gauge: Number of reminders in the current batch
+	// - UpDownCounter: Average reminders sent per batch
+	// - Histogram: reminder_last_sent time distribution
 	for _, repo := range repos {
-		logger.Debug().Str("repo", repo.ID.String()).
-			Time("previously", repo.ReminderLastSent.Time).Msg("updating reminder_last_sent")
-		err := r.store.UpdateReminderLastSentById(ctx, repo.ID)
+		repoReconcilerMessage, err := remindermessages.NewRepoReminderMessage(
+			repo.ProviderID, repo.RepoID, repo.ProjectID,
+		)
+		if err != nil {
+			errorSlice = append(errorSlice, err)
+			continue
+		}
+
+		logger.Debug().
+			Str("repo", repo.ID.String()).
+			Time("previously", repo.ReminderLastSent.Time).
+			Msg("updating reminder_last_sent")
+
+		err = qtx.UpdateReminderLastSentById(ctx, repo.ID)
 		if err != nil {
 			logger.Error().Err(err).Str("repo", repo.ID.String()).Msg("unable to update reminder_last_sent")
-			return []error{err}
+			errorSlice = append(errorSlice, err)
+			continue
+		}
+
+		messages = append(messages, repoReconcilerMessage)
+	}
+
+	if len(messages) != 0 {
+		err = r.eventPublisher.Publish(events.TopicQueueRepoReminder, messages...)
+		if err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("error publishing messages: %w", err))
+		} else {
+			logger.Info().Msgf("sent %d reminders", len(messages))
+
+			// Commit the transaction i.e update reminder_last_sent
+			// only if the messages were sent successfully
+			if err = tx.Commit(); err != nil {
+				logger.Error().Err(err).Msg("unable to commit transaction")
+				errorSlice = append(errorSlice, err)
+			}
```

Comment:
`UpdateReminderLastSentById` queries are part of a transaction and are only committed if the messages are published successfully. Still, the transaction could fail, which would result in reminders being sent, but the `reminder_last_sent` field won't be updated (better than false positive i.e. `reminder_last_sent` was updated but reminders weren't sent)

Reply:
It seems like the `reminder_last_sent` field is mostly diagnostic, because we're actually ticking forward on the (in-memory) repository cursor, rather than using `reminder_last_sent` to actually dictate when repos are reconsidered.  I think this is correct, but that suggests to me that strict update correctness is less necessary.

(I'm not arguing against keeping this data, but it seems like the actual mechanism that prevents re-evaluation is when the reminder is _received_ and a rule evaluation is complete, which could be minutes after the reminder is sent on a particularly bad day.)